### PR TITLE
feat(swap): Built Out Swap Input Components

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Built out Swap Input Components

Resolves <https://github.com/brave/brave-browser/issues/24157>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/180922711-e205db2a-5ef4-446b-916a-7a8ad1d27924.mov

### CODE REVIEW

File: app/src/components/buttons/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { StandardButton } from './standard-button'
+import { PresetButton } from './preset-button'
+
+export {
+  FlipTokensButton,
+  IconButton,
+  SelectTokenButton,
+  StandardButton,
+  PresetButton
+}
```
Comment (onyb) [inline app/src/components/buttons/index.ts:18]: @josheleonard to confirm, but I think we don't need to maintain these `index.ts` files anymore.

File: app/src/App.tsx [hunk 2]
```diff
@@ -35,14 +117,29 @@ function App () {
...
-        boxType='secondary'
+      <FlipTokensButton onClick={onClickFlipSwapTokens} />
+      <ToSection
+        getLocale={getLocale}
+        onClickSelectToken={onOpenSelectToken('to')}
+        token={toToken}
+        inputValue={toAmount}
+        onInputChange={handleOnSetToAmount}
+        hasInputError={false}
+        isLoading={isFetchingQuote}
```
Comment (onyb) [inline app/src/App.tsx:142]: Is it possible to add a `disabled` prop here? Solana swaps do not support modifying the TO amount yet, so we should disable this input, and ideally also have a way to visually indicate that we don't support editting it.

Ok with doing it in a different PR if you think we need product-design feedback first.

**Review by onyb — APPROVED** [review]
Looks great, with minor comments. Not a blocker for merging.

File: app/src/components/inputs/swap-input.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  onChange: (value: string) => void
+  hasError: boolean
+  value: string
+  autoFocus?: boolean
+}
+
+export const SwapInput = (props: Props) => {
+  const { onChange, value, autoFocus, hasError } = props
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply app/src/components/inputs/swap-input.tsx]: Done!

File: app/src/App.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply app/src/App.tsx:142]: Added this prop!
Will ping design and see if they want to modify anything when `disabled`.

File: app/src/components/buttons/index.ts (hunk 1 above)
Comment (Douglashdaniel) [reply app/src/components/buttons/index.ts:18]: We will need this file structure if we plan on making this a component library no?
Else we will have multiple imports to the `buttons` directory everywhere, etc..

Comment (josheleonard) [reply app/src/components/buttons/index.ts:18]: The repo name and current structure suggests that this project is a self-contained app containing the entire Swap experience (which could be loaded via iframe). If we are creating a library, are we exposing individual UI elements, or a single, self-contained Swap component?

These barrel files only make sense if we are exposing individual components for building a swap UI. Even then, a root-level index.ts should be the only file handling exposing our chosen library exports

Comment (josheleonard) [reply app/src/components/buttons/index.ts:18]: See: https://vitejs.dev/guide/build.html#library-mode for setting up the build step to output a library

Comment (Douglashdaniel) [reply app/src/components/buttons/index.ts:18]: Let's discuss this further and make a separate issue to address restructure if needed. 🙂

**Pull request merged**